hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/HazelcastClientCacheManager.java

```diff
@@ -18,56 +18,50 @@
 
 import com.hazelcast.cache.HazelcastCachingProvider;
 import com.hazelcast.cache.impl.AbstractHazelcastCacheManager;
-import com.hazelcast.cache.impl.CacheProxyUtil;
-import com.hazelcast.cache.impl.CacheService;
 import com.hazelcast.cache.impl.ICacheInternal;
 import com.hazelcast.cache.impl.nearcache.NearCacheManager;
 import com.hazelcast.client.impl.HazelcastClientInstanceImpl;
 import com.hazelcast.client.impl.HazelcastClientProxy;
-import com.hazelcast.client.impl.protocol.ClientMessage;
-import com.hazelcast.client.impl.protocol.codec.CacheCreateConfigCodec;
-import com.hazelcast.client.impl.protocol.codec.CacheGetConfigCodec;
-import com.hazelcast.client.impl.protocol.codec.CacheManagementConfigCodec;
-import com.hazelcast.client.spi.ClientContext;
-import com.hazelcast.client.spi.impl.ClientInvocation;
 import com.hazelcast.config.CacheConfig;
 import com.hazelcast.core.HazelcastInstance;
-import com.hazelcast.core.Member;
-import com.hazelcast.nio.Address;
-import com.hazelcast.nio.serialization.Data;
-import com.hazelcast.spi.serialization.SerializationService;
-import com.hazelcast.util.ExceptionUtil;
-import com.hazelcast.util.FutureUtil;
 
 import java.net.URI;
-import java.util.ArrayList;
-import java.util.Collection;
 import java.util.Properties;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
-import java.util.concurrent.Future;
-import java.util.concurrent.TimeUnit;
 
 import static com.hazelcast.util.Preconditions.checkNotNull;
 
 /**
+ * <p>
  * CacheManager implementation for client side
  * <p/>
  * Provides client side cacheManager functionality
  */
 public final class HazelcastClientCacheManager
         extends AbstractHazelcastCacheManager {
 
-    private final ClientContext clientContext;
+    private final HazelcastClientInstanceImpl client;
     private final ConcurrentMap<String, CacheConfig> configs = new ConcurrentHashMap<String, CacheConfig>();
 
     public HazelcastClientCacheManager(HazelcastClientCachingProvider cachingProvider, HazelcastInstance hazelcastInstance,
                                        URI uri, ClassLoader classLoader, Properties properties) {
         super(cachingProvider, hazelcastInstance, uri, classLoader, properties);
 
-        final ClientCacheDistributedObject setupRef = hazelcastInstance
-                .getDistributedObject(CacheService.SERVICE_NAME, "setupRef");
-        this.clientContext = setupRef.getClientContext();
+        /*
+         * TODO
```

Comment:
No. I have added these comments explicitly to mention that there might/should be better way(s) for accessing underlying client instance. But I have not introduced that way on this PR and postponed it by mentioning it with **TODO** comment.